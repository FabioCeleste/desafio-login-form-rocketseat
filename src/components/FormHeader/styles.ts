import styled, { css } from "styled-components";

import { theme } from "../../styles/theme";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.textTitle};

    h2 {
      margin-top: 3px;
      font-size: 16px;
      font-weight: 400;
    }
  `}
`;

export const Title = styled.h1`
  font-size: 26px;
  font-weight: 400;

  img {
    display: inline-block;
    vertical-align: middle;
  }

  span {
    margin-left: 8px;
    display: inline-block;
  }
`;
