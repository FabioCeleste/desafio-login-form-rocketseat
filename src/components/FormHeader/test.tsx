import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { FormHeader, FormHeaderProps } from ".";

const props: FormHeaderProps = {
  title: "Faça seu login",
  subTitle: "Entre com suas informações de cadastro",
};

describe("<FormHeader />", () => {
  it("should render a header login", () => {
    renderTheme(<FormHeader {...props} />);

    expect(
      screen.getByRole("heading", { name: props.title })
    ).toBeInTheDocument();
    expect(
      screen.getByRole("heading", { name: props.subTitle })
    ).toBeInTheDocument();
    expect(screen.getByRole("img")).toHaveAttribute(
      "src",
      "/assets/images/log-in.svg"
    );
  });

  it("should match snapshot", () => {
    const { container } = renderTheme(<FormHeader {...props} />);

    expect(container).toMatchSnapshot();
  });
});
