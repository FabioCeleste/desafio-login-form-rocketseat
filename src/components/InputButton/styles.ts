import styled, { css } from "styled-components";
import { theme } from "../../styles/theme";

export const Wrapper = styled.button`
  width: 100%;
  height: 50px;
  margin-top: 20px;

  background: ${theme.colors.primaryColor};
  color: ${theme.colors.background};
  font-size: 18px;
  font-weight: 600;
  border: none;
  border-radius: 4px;
`;
