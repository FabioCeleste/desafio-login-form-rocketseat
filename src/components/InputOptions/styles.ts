import styled, { css } from "styled-components";
import { theme } from "../../styles/theme";

export type RememberMeProps = {
  shouldRemember: boolean;
};

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  margin-top: 20px;

  font-weight: bold;
  color: ${theme.colors.textBase};

  label {
    display: inline-block;
    padding-right: 10px;
    white-space: nowrap;
  }

  .spanRemember {
    vertical-align: middle;
    padding-left: 10px;
  }
`;

export const RememberMe = styled.input<RememberMeProps>`
  ${({ shouldRemember }) => css`
    width: 20px;
    height: 20px;
    border: 1px solid ${theme.colors.textBase};

    vertical-align: middle;

    border-radius: 3px;
    -moz-appearance: none;
    -webkit-appearance: none;
    -o-appearance: none;
    cursor: pointer;
    ${shouldRemember ? `background: ${theme.colors.primaryColor};` : ""}
  `}
`;

// export const Checkmark = styled.span<RememberMeProps>`
//   ${({ shouldRemember }) => css`
//     color: ${theme.colors.background};
//     position: relative;
//     cursor: pointer;
//     right: 16px;
//     top: 1.5px;

//     ${shouldRemember ? "" : "display: none;"}
//   `}
// `;

export const PasswordRecovery = styled.div`
  a {
    color: ${theme.colors.primaryColor};
    text-decoration: none;
  }
  font-weight: 400;
`;
