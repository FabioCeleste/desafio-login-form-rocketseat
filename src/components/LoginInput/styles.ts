import styled, { css } from "styled-components";

import { theme } from "../../styles/theme";

export interface InputProps {
  inputSvg: string;
  focusSvg: string;
}

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: 20px;
`;

export const Label = styled.label`
  color: ${theme.colors.textBase};
  font-size: 16px;
`;

export const Input = styled.input<InputProps>`
  ${({ inputSvg, focusSvg }) => css`
    background: url(${inputSvg});
    background-repeat: no-repeat;
    background-position-y: center;
    background-position-x: 3px;

    color: ${theme.colors.textTitle};
    ::placeholder {
      color: ${theme.colors.textBase};
    }

    padding-left: 30px;
    border: 1px solid ${theme.colors.textTitle};
    border-radius: 5px;

    height: 36px;
    width: 335px;

    &:focus {
      border: 3px solid ${theme.colors.primaryColor};
      outline: none;

      background: url(${focusSvg});
      background-repeat: no-repeat;
      background-position-y: center;
      background-position-x: 3px;
    }
  `}
`;
