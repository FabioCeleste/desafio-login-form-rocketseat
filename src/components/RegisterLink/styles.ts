import styled, { css } from "styled-components";
import { theme } from "../../styles/theme";

export const Wrapper = styled.div`
  display: flex;
  align-item: center;
  justify-content: center;

  margin-top: 14px;

  a {
    color: ${theme.colors.primaryColor};
    text-decoration: none;
  }

  .register-span {
    font-weight: bold;
  }
`;
