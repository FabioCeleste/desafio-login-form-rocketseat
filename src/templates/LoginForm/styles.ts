import styled from "styled-components";

import { theme } from "../../styles/theme";

export const Wrapper = styled.div`
  display: flex;
  width: 100vw;
  height: 100vh;
`;

export const Main = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  height: 100vh;
  width: 50%;
  background: ${theme.colors.background};

  @media (max-width: 800px) {
    width: 100vw;
    background: ${theme.colors.formBackground};
    overflow: hidden;
  }
`;

export const LogoIcon = styled.img`
  width: 115px;
  height: 28px;
  align-self: flex-start;

  margin-left: 40px;
  margin-top: 40px;
`;

export const FormDiv = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  padding: 55px 120px;

  background: ${theme.colors.formBackground};
  margin-top: 126px;
`;

export const Form = styled.form``;

export const Aside = styled.div`
  @media (max-width: 800px) {
    display: none;
  }
`;

export const Image = styled.img`
  height: 100vh;
  width: 50vw;
`;
