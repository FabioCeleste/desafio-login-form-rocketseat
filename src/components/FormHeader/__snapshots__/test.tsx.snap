// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormHeader /> should match snapshot 1`] = `
.c0 {
  color: #d4ccb6;
}

.c0 h2 {
  margin-top: 3px;
  font-size: 16px;
  font-weight: 400;
}

.c1 {
  font-size: 26px;
  font-weight: 400;
}

.c1 img {
  display: inline-block;
  vertical-align: middle;
}

.c1 span {
  margin-left: 8px;
  display: inline-block;
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1"
    >
      <img
        src="/assets/images/log-in.svg"
      />
      <span>
        Faça seu login
      </span>
    </h1>
    <h2>
      Entre com suas informações de cadastro
    </h2>
  </div>
</div>
`;
